### PR TITLE
Added section on controlling node selection

### PR DESCRIPTION
**Pull request by alexmilowski**
Added a section under "Getting Started with Redis Enterprise Software using Kubernetes" called "Controlling Node Selection" that covers node selectors, node pools, pod anti-affinity, and rack awareness.

### CODE REVIEW

File: content/platforms/kubernetes/pod-scheduling.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        app: redis-enterprise
+        redis.io/cluster: rec
+        redis.io/role: node
+    topologyKey: kubernetes.io/hostname
+```
+
+Each pod has the three labels above where `redis.io/cluster` is the name of your
+cluster.
+
+You can override this rule to restrict or widen the nodes on which Redis Enterprise
```
Comment (amiramm) [inline content/platforms/kubernetes/pod-scheduling.md:99]: @alexmilowski are you sure Operator won't just override as it controls the StatefulSet? I would not have customers touch nodeAffinity

File: content/platforms/kubernetes/pod-scheduling.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+metadata:
+  name: example-redisenterprisecluster
+spec:
+  size: 3
+  rackAwarenessNodeLabel: failure-domain.beta.kubernetes.io/zone
+```
+
+The operator will change the `topologyKey` for the pod anti-affinity rule to
+the label value used. Combining `rackAwarenessNodeLabel` and `podAntiAffinity`
+requires care to ensure you make the same change.
```
Comment (amiramm) [inline content/platforms/kubernetes/pod-scheduling.md:187]: Need to add a section around tolerations which will be introduced in 6.0. 
The primary use-case being running Redis Enterprise on a dedicated set of nodes.
https://redislabs.atlassian.net/browse/RED-33069

File: content/platforms/kubernetes/pod-scheduling.md (hunk 1 above)
Comment (alexmilowski) [reply content/platforms/kubernetes/pod-scheduling.md:99]: The StatefulSet for the cluster has a `podAntiAffinity` that is generated. What is specified in the spec is copied verbatim [here in resources.go](https://github.com/RedisLabs/redis-enterprise-operator/blob/ee408af23f757bceca350983864954e2266d018c/pkg/controller/redisenterprisecluster/resources.go#L585). It is never updated unless the CR is updated. There does not appear to be any override except that `podAntiAffinity` and `antiAffinityAdditionalTopologyKeys` cannot both be set.

File: content/platforms/kubernetes/pod-scheduling.md (hunk 2 above)
Comment (alexmilowski) [reply content/platforms/kubernetes/pod-scheduling.md:229]: I have added a section on the use of the `podTolerations` property.

**Comment by amiramm** [comment]
@bmansheim reviewed, please merge.

**Comment by bmansheim** [comment]
@alexmilowski Please check my edits.

File: content/platforms/kubernetes/pod-scheduling.md [hunk 3]
```diff
@@ -181,38 +168,31 @@ spec:
...
-is increased in the event of partitions or other rack (or region) related
-failures.
+You can configure Redis Enterprise with rack-zone awareness to increase availability
+during partitions or other rack (or region) related failures.
 
-Rack awareness is a single property in the Redis Enterprise cluster CRD named
-`rackAwarenessNodeLabel` whose value is the rack label applied to the node. This
-label is often `failure-domain.beta.kubernetes.io/zone` as documented in
+Rack-zone awareness is a single property in the Redis Enterprise cluster CRD named `rackAwarenessNodeLabel`.
+This value for this label is commonly `failure-domain.beta.kubernetes.io/zone` as documented in
```
Comment (alexmilowski) [inline content/platforms/kubernetes/pod-scheduling.md:180]: okaty

File: content/platforms/kubernetes/pod-scheduling.md [hunk 4]
```diff
@@ -181,38 +168,31 @@ spec:
...
 ['Running in multiple zones'](https://kubernetes.io/docs/setup/best-practices/multiple-zones/#nodes-are-labeled).
 
-You can check your nodes are labeled via:
+You can check the value for this label in your nodes with the command:
```
Comment (alexmilowski) [inline content/platforms/kubernetes/pod-scheduling.md:183]: okay.

File: content/platforms/kubernetes/pod-scheduling.md [hunk 5]
```diff
@@ -181,38 +168,31 @@ spec:
...
-```sh
 name                                            rack\zone
 ip-10-0-x-a.eu-central-1.compute.internal    eu-central-1a
 ip-10-0-x-b.eu-central-1.compute.internal    eu-central-1a
 ip-10-0-x-c.eu-central-1.compute.internal    eu-central-1b
 ip-10-0-x-d.eu-central-1.compute.internal    eu-central-1b
 ```
 
-All you need to do is tell the operator which label represents the rack zone:
+You can configure the CRD to use another value to represent the rack-zone:
```
Comment (alexmilowski) [inline content/platforms/kubernetes/pod-scheduling.md:195]: You must specify a value. This is not specified by default.

"You can configure the node label to read for the rack zone by setting the `rackAwarenessNodeLabel` property:"

File: content/platforms/kubernetes/pod-scheduling.md [hunk 6]
```diff
@@ -224,6 +204,7 @@ spec:
   rackAwarenessNodeLabel: failure-domain.beta.kubernetes.io/zone
 ```
 
-The operator will change the `topologyKey` for the pod anti-affinity rule to
-the label value used. Combining `rackAwarenessNodeLabel` and `podAntiAffinity`
-requires care to ensure you make the same change.
+{{% note %}}
+When you use the `rackAwarenessNodeLabel`, the operator changes the `topologyKey` for the pod anti-affinity rule to the label value used.
+If you use `rackAwarenessNodeLabel` and `podAntiAffinity` together, make sure that the resulting `topologyKey` is as expected.
+{{% /note %}}
```
Comment (alexmilowski) [inline content/platforms/kubernetes/pod-scheduling.md:210]: No. It does not change the topologyKey if you have changed the anti-affinity rules as well. You have to do that yourself.

"When you use the `rackAwarenessNodeLabel`, the operator will change the `topologyKey` for the anti-affinity rule to the label value used unless you have specified the `podAntiAffinity` as well. If you use `rackAwarenessNodeLabel` and `podAntiAffinity` tougher, you must make sure that the `topologyKey` in your pod anti-affinity rule is set to the node label."

File: content/platforms/kubernetes/pod-scheduling.md [hunk 7]
```diff
@@ -47,24 +47,21 @@ spec:
...
 
 
 ## Using node pools
 
-A node pool is a typically part of the underlying infrastructure of the K8s cluster
-deployment and provider. Often, node pools are similarly-configured classes of
-nodes (e.g., nodes with the same allocated amount of memory and CPU). Implementors
-will often label these nodes with a consistent set of labels.
+A node pool is a common part of the underlying infrastructure of the Kubernetes cluster deployment and provider.
+Often, node pools are similarly-configured classes of nodes, for example nodes with the same allocated amount of memory and CPU).
```
[the remark this answers is not in the document]
Comment (alexmilowski) [reply content/platforms/kubernetes/pod-scheduling.md]: fixed

**Pull request merged**